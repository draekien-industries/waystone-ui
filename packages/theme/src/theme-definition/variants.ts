export const variants = {
  text: {
    default: {
      fontSize: 'body',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
      marginY: 'small',
    },
    hero: {
      fontSize: 'hero',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'large',
    },
    heading: {
      fontSize: 'heading',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'medium',
    },
    title: {
      fontSize: 'title',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'small',
    },
    subtitle: {
      fontSize: 'subtitle',
      fontWeight: 'semibold',
      lineHeight: 'heading',
      fontFamily: 'body',
      marginY: 'small',
    },
    eyebrow: {
      fontSize: 'body',
      fontWeight: 'medium',
      lineHeight: 'body',
      fontFamily: 'body',
      textTransform: 'uppercase',
      letterSpacing: 'uppercase',
    },
    label: {
      fontSize: 'body',
      fontWeight: 'medium',
      lineHeight: 'body',
      fontFamily: 'body',
      marginY: 0,
    },
    body: {
      fontSize: 'body',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
      marginY: 'small',
    },
    caption: {
      fontSize: 'caption',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
    },
    small: {
      fontSize: 'small',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
    },
    monospace: {
      fontFamily: 'monospace',
      lineHeight: 'body',
      fontWeight: 'regular',
      fontSize: 'body',
    },
  },
} as const;
