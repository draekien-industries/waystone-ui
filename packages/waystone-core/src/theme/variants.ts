import { ThemeUIStyleObject } from 'theme-ui';

export const variants: Record<string, ThemeUIStyleObject> = {
  text: {
    default: {
      fontSize: 'body',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
      marginY: 'sm',
    },
    hero: {
      fontSize: 'hero',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'lg',
    },
    heading: {
      fontSize: 'heading',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'md',
    },
    title: {
      fontSize: 'title',
      fontWeight: 'bold',
      lineHeight: 'heading',
      fontFamily: 'heading',
      marginY: 'sm',
    },
    subtitle: {
      fontSize: 'subtitle',
      fontWeight: 'semibold',
      lineHeight: 'heading',
      fontFamily: 'body',
      marginY: 'sm',
    },
    eyebrow: {
      fontSize: 'body',
      fontWeight: 'medium',
      lineHeight: 'body',
      fontFamily: 'body',
      textTransform: 'uppercase',
      letterSpacing: 'uppercase',
    },
    body: {
      fontSize: 'body',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
      marginY: 'sm',
    },
    caption: {
      fontSize: 'caption',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
    },
    small: {
      fontSize: 'small',
      fontWeight: 'regular',
      lineHeight: 'body',
      fontFamily: 'body',
    },
    monospace: {
      fontFamily: 'monospace',
      lineHeight: 'body',
      fontWeight: 'regular',
      fontSize: 'body',
    },
  },
};

export default variants;
