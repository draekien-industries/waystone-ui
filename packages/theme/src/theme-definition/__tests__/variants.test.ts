import { variants } from '../variants';

describe('variants', () => {
  it('should match snapshot', () => {
    expect(variants).toMatchSnapshot();
  });
});
