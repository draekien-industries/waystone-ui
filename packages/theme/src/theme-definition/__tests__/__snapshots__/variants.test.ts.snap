// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variants should match snapshot 1`] = `
{
  "text": {
    "body": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
      "marginY": "small",
    },
    "caption": {
      "fontFamily": "body",
      "fontSize": "caption",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "default": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
      "marginY": "small",
    },
    "eyebrow": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "medium",
      "letterSpacing": "uppercase",
      "lineHeight": "body",
      "textTransform": "uppercase",
    },
    "heading": {
      "fontFamily": "heading",
      "fontSize": "heading",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "medium",
    },
    "hero": {
      "fontFamily": "heading",
      "fontSize": "hero",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "large",
    },
    "label": {
      "fontFamily": "body",
      "fontSize": "body",
      "fontWeight": "medium",
      "lineHeight": "body",
      "marginY": 0,
    },
    "monospace": {
      "fontFamily": "monospace",
      "fontSize": "body",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "small": {
      "fontFamily": "body",
      "fontSize": "small",
      "fontWeight": "regular",
      "lineHeight": "body",
    },
    "subtitle": {
      "fontFamily": "body",
      "fontSize": "subtitle",
      "fontWeight": "semibold",
      "lineHeight": "heading",
      "marginY": "small",
    },
    "title": {
      "fontFamily": "heading",
      "fontSize": "title",
      "fontWeight": "bold",
      "lineHeight": "heading",
      "marginY": "small",
    },
  },
}
`;
